### PR TITLE
Remove bitslice.rs

### PR DESCRIPTION
**Pull request by nnethercote**
As the comment in `bitslice.rs` says:

> FIXME: merge with `bitvec`

### CODE REVIEW

**Comment by nnethercote** [comment]
Local benchmarking runs showed that this change had no effect on performance.

File: src/librustc_data_structures/indexed_set.rs [hunk 1]
```diff
@@ -196,9 +140,7 @@ impl<T: Idx> IdxSet<T> {
 
     pub fn iter(&self) -> Iter<T> {
         Iter {
-            cur: None,
-            iter: self.words().iter().enumerate(),
-            _pd: PhantomData,
+            iter: self.0.iter()
```
Comment (Mark-Simulacrum) [inline src/librustc_data_structures/indexed_set.rs]: Is there any point in the wrapper here? Technically it hides possible optimizations (TrustedLen, DoubleEnded, etc) that might be implemented

**Comment by nnethercote** [comment]
@Mark-Simulacrum: I fixed the compile errors.

**Comment by nnethercote** [comment]
r=Mark-Simulacrum

**Pull request merged**